contracts/dao/QuadraticTimelockedSubdelegator.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./ITimelockedDelegator.sol";
+import "../utils/timelock/QuadraticTokenTimelock.sol";
+
+/// @title a proxy delegate contract for TRIBE
+/// @author Fei Protocol
+contract Delegatee is Ownable {
+    ITribe public tribe;
+
+    /// @notice Delegatee constructor
+    /// @param _delegatee the address to delegate TRIBE to
+    /// @param _tribe the TRIBE token address
+    constructor(address _delegatee, address _tribe) {
+        tribe = ITribe(_tribe);
+        tribe.delegate(_delegatee);
+    }
+
+    /// @notice send TRIBE back to timelock and selfdestruct
+    function withdraw() public onlyOwner {
+        ITribe _tribe = tribe;
+        uint256 balance = _tribe.balanceOf(address(this));
+        _tribe.transfer(owner(), balance);
+        selfdestruct(payable(owner()));
+    }
+}
+
+/// @title a timelock for TRIBE allowing for sub-delegation
+/// @author Fei Protocol
+/// @notice allows the timelocked TRIBE to be delegated by the beneficiary while locked
+contract QuadtraticTimelockedSubdelegator is ITimelockedDelegator, QuadraticTokenTimelock {
+    /// @notice associated delegate proxy contract for a delegatee
+    mapping(address => address) public override delegateContract;
+
+    /// @notice associated delegated amount of TRIBE for a delegatee
+    /// @dev Using as source of truth to prevent accounting errors by transferring to Delegate contracts
+    mapping(address => uint256) public override delegateAmount;
+
+    /// @notice the TRIBE token contract
+    ITribe public override tribe;
+
+    /// @notice the total delegated amount of TRIBE
+    uint256 public override totalDelegated;
+
+    /// @notice Delegatee constructor
+    /// @param _beneficiary default delegate, admin, and timelock beneficiary
+    /// @param _duration duration of the token timelock window
+    /// @param _tribe the TRIBE token address
+    /// @param _cliff the seconds before first claim is allowed
+    /// @param _clawbackAdmin the address which can trigger a clawback
+    /// @param _startTime the initial time to use for timelock
+    constructor(
+        address _beneficiary,
+        uint256 _duration,
+        address _tribe,
+        uint256 _cliff,
+        address _clawbackAdmin,
+        uint256 _startTime
+    ) QuadraticTokenTimelock(_beneficiary, _duration, _tribe, _cliff, _clawbackAdmin, _startTime) {
+        tribe = ITribe(_tribe);
```

Comment:
Note that if we set this timelock up for someone, they could delegate all of their tokens to another address. The TRIBE gets transferred to a new Delegatee contract, so it is then out of reach from clawback.

Maybe add a require(_clawbackAdmin == address(0)) in the constructor, to make sure this kind of timelock is only used for people that don't have a clawback.

Reply:
The Rari team subdelegator does have a clawbackAdmin, so I think you should override the clawback function here to recall tokens from all delegatees

Reply:
Fixing this, enforcing clawback admin must be 0